kernel/kernel-api/src/main/java/io/delta/kernel/types/StructField.java

```diff
@@ -59,20 +56,20 @@ public class StructField {
     private final String name;
     private final DataType dataType;
     private final boolean nullable;
-    private final Map<String, String> metadata;
+    private final FieldMetadata metadata;
 
     public StructField(
             String name,
             DataType dataType,
             boolean nullable) {
-        this(name, dataType, nullable, Collections.emptyMap());
+        this(name, dataType, nullable, FieldMetadata.builder().build());
```

Comment:
why not have a `FieldMetadata.empty()` API for this?

Reply:
Yep good idea added